plugins/generator-1.18.2/forge-1.18.2/features/block_predicate_matching_blocks.java.ftl

```diff
@@ -0,0 +1,7 @@
+<#-- Hacky workarounds to allow this predicate in 1.18 -->
+<#if input$blockSet?contains("BlockTags")> <#-- The holder set is a tag, we "extract" the tag part from the input -->
+    BlockPredicate.matchesTag(${input$blockSet?remove_beginning("Registry.BLOCK.getTag(")?remove_ending(".orElseThrow())")}
+<#else> <#-- The holder set is a block list, we "extract" the blocks from the input -->
+    BlockPredicate.matchesBlocks(List.of(${input$blockSet?remove_beginning("HolderSet.direct(Block::builtInRegistryHolder")?remove_beginning(",")?remove_ending(")")})

```

Comment:
Why is the part

`HolderSet.direct(Block::builtInRegistryHolder")`

added in block_holder_set_list and then removed here. Maybe it would make more sense to just provide come separated list in block_holder_set_list and then wrap it with 

`HolderSet.direct(Block::builtInRegistryHolder")`

code where needed.

---------------

Instead of `input$blockSet?contains("BlockTags")` you could also use `/*@*/` tag specifier

Reply:
Fair enough